frontend/src/components/common/ReleaseNotes/UpdatePopup.tsx

```diff
@@ -2,11 +2,57 @@ import { Button, Snackbar } from '@material-ui/core';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 
-function UpdatePopup(props: { releaseDownloadURL: string }) {
+function UpdatePopup(props: {
+  releaseDownloadURL?: string | null;
+  fetchingRelease?: boolean;
+  releaseFetchFailed?: boolean;
+  skipUpdateCheck?: boolean;
+}) {
   const [show, setShow] = React.useState(true);
-  const { releaseDownloadURL } = props;
+  const { releaseDownloadURL, fetchingRelease, releaseFetchFailed, skipUpdateCheck } = props;
   const { t } = useTranslation('frequent');
 
+  if (fetchingRelease) {
+    return (
+      <Snackbar
+        open={show}
+        message={t('release|fetching release...')}
+        ContentProps={{
+          'aria-describedby': t('release|fetching release message'),
+        }}
+      />
```

Comment:
This should have a skip button, because if it's taking too long, the user may want to skip fetching the info.

Reply:
It should also probably be delayed for 1/2 a sec (to give time for the release just to be checked without noise for the user) and hidden once the release info request is finished.

Reply:
There is a skip/dismiss button now. However I don't see where the "delay for 1/2 a sec" is.